### PR TITLE
#130 - Do not depend on PIL on import of whylogs

### LINKED ISSUE #130: whylogs mainline requires Pillow as a hard dependency
Trying to run whylogs for Pandas and getting this:
```
  File "/miniconda/envs/custom_env/lib/python3.7/site-packages/whylogs/__init__.py", line 2, in <module>
    from .app.config import SessionConfig, WriterConfig
  File "/miniconda/envs/custom_env/lib/python3.7/site-packages/whylogs/app/__init__.py", line 5, in <module>
    from .session import SessionConfig
  File "/miniconda/envs/custom_env/lib/python3.7/site-packages/whylogs/app/session.py", line 12, in <module>
    from whylogs.app.logger import Logger
  File "/miniconda/envs/custom_env/lib/python3.7/site-packages/whylogs/app/logger.py", line 10, in <module>
    from whylogs.app.writers import Writer
  File "/miniconda/envs/custom_env/lib/python3.7/site-packages/whylogs/app/writers.py", line 15, in <module>
    from whylogs.core import DatasetProfile
  File "/miniconda/envs/custom_env/lib/python3.7/site-packages/whylogs/core/__init__.py", line 8, in <module>
    from .image_profiling import TrackImage, _METADATA_DEFAULT_ATTRIBUTES
  File "/miniconda/envs/custom_env/lib/python3.7/site-packages/whylogs/core/image_profiling.py", line 2, in <module>
    from PIL.Image import Image as ImageType
ModuleNotFoundError: No module named 'PIL'
```

### PR DESCRIPTION
**Pull request by andyndang**
* Move PIL imports into local import (#130)
* Rename cache to cache_size to reflect what it actually represents
* Add some minor documentation

Tested by importing whylogs in an environment without PIL

### CODE REVIEW

File: src/whylogs/app/logger.py [hunk 1]
```diff
@@ -73,7 +70,7 @@ def __init__(self,
         self.dataset_name = dataset_name
         self.writers = writers
         self.verbose = verbose
-        self.cache = cache
+        self.cache = cache_size
```
Comment (lalmei) [inline src/whylogs/app/logger.py]: I would keep the same name, either direction.

File: src/whylogs/app/logger.py [hunk 2]
```diff
@@ -355,46 +357,51 @@ def log_image(self, image, feature_transforms: Optional[List[Callable]] = None,
         track_image(self._profiles[-1]["full_profile"])
 
     def log_csv(self,
-                filepath_or_buffer: FilePathOrBuffer,
-                segments: Optional[Union[List[SegmentTags], List[str]]] = None,
-                profile_full_dataset: bool = False, **kwargs,):
+                filepath_or_buffer: Union[str, Path, IO[AnyStr]],
+                segment_tags: Optional[Union[List[SegmentTags], List[str]]] = None,
```
Comment (lalmei) [inline src/whylogs/app/logger.py]: A segment is made up of multiple sets of segment_tags.

File: src/whylogs/app/logger.py [hunk 3]
```diff
@@ -355,46 +357,51 @@ def log_image(self, image, feature_transforms: Optional[List[Callable]] = None,
...
             return None
         if self.should_rotate():
             self._rotate_time()
 
-        # segmnet check  in case segments are just keys
+        # segment check  in case segments are just keys
         self.profile_full_dataset = profile_full_dataset
-        if (segments is not None):
-            self.set_segments(segments)
+        if segment_tags is not None:
```
Comment (lalmei) [inline src/whylogs/app/logger.py]: would revert to segments

File: src/whylogs/app/session.py [hunk 4]
```diff
@@ -72,10 +72,10 @@ def logger(
         session_timestamp: Optional[datetime.datetime] = None,
         tags: Dict[str, str] = {},
         metadata: Dict[str, str] = None,
-        segments: Optional[Union[List[Dict], List[str]]] = None,
+        segment_tags: Optional[Union[List[Dict], List[str]]] = None,
```
Comment (lalmei) [inline src/whylogs/app/session.py]: same as above

File: src/whylogs/core/image_profiling.py [hunk 5]
```diff
@@ -58,7 +50,7 @@ class TrackImage:
 
     def __init__(self,
                  filepath: str = None,
-                 img: ImageType = None,
+                 img=None,
```
Comment (lalmei) [inline src/whylogs/core/image_profiling.py]: Type checked removed why ?

**Review by lalmei — CHANGES_REQUESTED** [review]

File: src/whylogs/core/image_profiling.py (hunk 5 above)
Comment (andyndang) [reply src/whylogs/core/image_profiling.py]: hard import on PIL :). This is preventing `import whylogs` (see the attached issue)

File: src/whylogs/app/logger.py (hunk 2 above)
Comment (andyndang) [reply src/whylogs/app/logger.py]: A group of segment tags is a segment (not `segments` in plural form).

Either we create a wrapper object for segment tags and call it `Segment`. I don't like having a `list` parameter but calling it singular - it's very confusing from API design perspective.

File: src/whylogs/app/logger.py (hunk 3 above)
Comment (andyndang) [reply src/whylogs/app/logger.py]: See comment above

File: src/whylogs/app/logger.py (hunk 1 above)
Comment (andyndang) [reply src/whylogs/app/logger.py]: Thanks. I missed this

File: src/whylogs/core/image_profiling.py (hunk 5 above)
Comment (andyndang) [reply src/whylogs/core/image_profiling.py]: Using try catch and now we can keep the type :)

File: src/whylogs/app/logger.py (hunk 2 above)
Comment (lalmei) [reply src/whylogs/app/logger.py]: yes, but the user needs to define multiple segments.

Comment (lalmei) [reply src/whylogs/app/logger.py]: This is so we can have overlapping segments.

Comment (andyndang) [reply src/whylogs/app/logger.py]: A segment = List[SegmentTags]

In order to support multple segments, the API must accept `List[List[SegmentTags]]`. Maybe I'm missing some context with the API here?

Comment (andyndang) [reply src/whylogs/app/logger.py]: My bad - misunderstood `SegmentTags` (plural) vs `SegmentTag`. Renamed `SegmentTags` to `Segment` to make it clear that it represents a Segment

**Review by lalmei — APPROVED** [review]
LGTM

**Pull request merged**